go/example.go

```diff
@@ -96,47 +101,53 @@ func main() {
 	client, err := flight.NewClientWithMiddleware(
 		net.JoinHostPort(config.Host, config.Port),
 		nil,
-		[]flight.ClientMiddleware{
-			flight.NewClientCookieMiddleware(),
-		},
+		[]flight.ClientMiddleware{flight.NewClientCookieMiddleware()},
 		grpc.WithTransportCredentials(creds),
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer client.Close()
 
+	if err := run(config, client, WrapRecordReader); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(config interfaces.FlightConfig, client flight.Client,
+	readerCreator func(flight.FlightService_DoGetClient) (RecordReader, error),
+) error {
+
 	// Two WLM settings can be provided upon initial authentication with the dremio
 	// server flight endpoint:
 	//  - routing-tag
 	//  - routing-queue
 	ctx := metadata.NewOutgoingContext(context.TODO(),
```

Comment:
Do we have this somewhere in the docs? 
In golang in general the context should be passed outside. For now it's fine to keep it here, but it would be great to add docs link where this is described.